File: Javascript/Find-The-Town-Judge.js
/**
 * @param {number} n
 * @param {number[][]} trust
 * @return {number}
 */
var findJudge = function (n, trust) {
  const trustedBy = Array(n + 1).fill(0);
  for (const [a, b] of trust) {
    trustedBy[a] = NaN; // since the judge doesn't trust anyone, a[i] can't be the judge
    trustedBy[b]++; // ++ applied to NaN is still NaN, and won't amount to n-1
  }
  // Because trust pairs are unique, there can be only one judge, or none at all.
  return trustedBy.indexOf(n - 1) || 1;
};

File: Go/Find-The-Town-Judge.go
package main



func findJudge(n int, trust [][]int) int {
	if n == 1 {
		return 1
	}
	
	trustedBy := make([]int, n+1)

	for _, tru := range trust {
		a, b := tru[0], tru[1]
		trustedBy[a] -= 1
		trustedBy[b] += 1
	}

	for i, v := range trustedBy {
		if v == n-1 {
			return i
		}
	}
	return -1
}
